vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow/src/main/resources/META-INF/resources/frontend/vaadin-spreadsheet/vaadin-spreadsheet.js

```diff
@@ -404,6 +404,10 @@ export class VaadinSpreadsheet extends LitElement {
     this.api.editCellComment(col, row);
   }
 
+  onPopupButtonOpen(row, column, contentId) {
+    requestAnimationFrame(() => this.api.onPopupButtonOpened(row, column, contentId));
```

Comment:
This was needed to work around the case when a `PopupButton` is created and opened from the server on the same roundtrip. It seems that calls to `setProperty` are made after `callJsFunction` even when the former is called before the later. When it happens, `onPopupButtonOpen` is called before the `PopupButtonWidget` is created.

Reply:
I think it's better not do this in the WC and do in the API, take a look to the `api.editCellComment` then all workarounds for the same problematic are in the same class, and using the same approach

Reply:
With the new approach, this is no longer needed.